package/src/components/Channel/Channel.tsx

```diff
@@ -690,50 +669,144 @@ const ChannelWithContext = <
+        copyChannelState();
+      }
+    }
+  };
+
   useEffect(() => {
+    let listener: ReturnType<typeof channel.on>;
     const initChannel = async () => {
       if (!channel || !shouldSyncChannel || channel.offlineMode) return;
-      /**
-       * Loading channel at first unread message  requires channel to be initialized in the first place,
-       * since we use read state on channel to decide what offset to load channel at.
-       * Also there is no use case from UX perspective, why one would need loading uninitialized channel at particular message.
-       * If the channel is not initiated, then we need to do channel.watch, which is more expensive for backend than channel.query.
-       */
-      if (!channel.initialized) {
-        await loadChannel();
+      let errored = false;
+
+      if (!channel.initialized || !channel.state.isUpToDate) {
+        try {
+          await channel?.watch();
+        } catch (err) {
+          console.warn('Channel watch request failed with error:', err);
+          setError(true);
+          errored = true;
+        }
       }
 
-      if (messageId) {
-        loadChannelAroundMessage({ messageId });
+      if (!errored) {
+        initStateFromChannel(channel);
+        loadInitialMessagesStateFromChannel(channel, channel.state.messagePagination.hasPrev);
       }
-      // The condition, where if the count of unread messages is greater than 4, then scroll to the first unread message.
-      else if (
+
+      if (messageId) {
+        await loadChannelAroundMessage({ messageId, setTargetedMessage });
+      } else if (
         initialScrollToFirstUnreadMessage &&
         channel.countUnread() > scrollToFirstUnreadThreshold
       ) {
-        loadChannelAtFirstUnreadMessage();
-      }
-      // If the messageId is undefined and the last message and the current message id do not match we load the channel at the very bottom.
-      else if (
-        channel.state.messages?.[channel.state.messages.length - 1]?.id !==
-          channel.state.latestMessages?.[channel.state.latestMessages.length - 1]?.id &&
-        !messageId
-      ) {
-        await loadChannel();
+        await loadChannelAtFirstUnreadMessage({ setTargetedMessage });
       }
+      listener = channel.on(handleEvent);
     };
 
     initChannel();
 
     return () => {
       copyChannelState.cancel();
-      copyReadState.cancel();
-      copyTypingState.cancel();
-      loadMoreFinished.cancel();
       loadMoreThreadFinished.cancel();
+      listener?.unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [channel.cid, messageId, shouldSyncChannel]);
+
+  // subscribe to channel.deleted event
+  useEffect(() => {
+    const { unsubscribe } = client.on('channel.deleted', (event) => {
+      if (event.cid === channel?.cid) {
+        setDeleted(true);
+      }
+    });
+
+    return unsubscribe;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [channelId]);
```

Comment:
Please make these dependencies and the ones below it stable and follow rules of hooks

Reply:
Cool, I have done for a few of them.